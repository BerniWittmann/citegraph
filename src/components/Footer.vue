<template>
  <el-row class="footer-wrap">
    <el-col :span="24" class="footer-col">
      <el-row class="footer">
        <el-col :lg="12" :md="12" :sm="12" :xs="24">
          <p class="footer-paragraph">
            <router-link :to="{ name: 'home' }">{{ $t('footer.home') }}</router-link>
          </p>
          <p class="footer-paragraph">
            <router-link :to="{ name: 'about' }">{{ $t('footer.about') }}</router-link>
          </p>
        </el-col>
        <el-col :lg="{span: 7, offset: 5}" :md="{span: 7, offset: 5}" :sm="{span: 7, offset: 5}"
                :xs="{span: 24}">
          <p class="footer-paragraph">
            {{ $t('footer.version', { version: version }) }}
          </p>
        </el-col>
      </el-row>
    </el-col>
  </el-row>
</template>

<style scoped lang="scss">
.footer-wrap {
  flex: 1;
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  & > div {
    display: flex;
    justify-content: center;
    flex-direction: column;
    & > a {
      align-self: flex-start
    }
  }
  .footer {
    padding: 30px 20vw;
    background: $footer_bg_color;
    color: $footer_text_color;
    margin: 0;
    .footer-paragraph {
      font-size: 14px;
      &, p {
        color: inherit;
        -webkit-margin-before: 0;
        -webkit-margin-after: 0.4em;
      }
      &, a {
        &, &:hover, &:visited, &:focus, &:active {
          color: inherit;
          text-decoration: none;
        }
        -webkit-margin-before: 0;
        -webkit-margin-after: 0.4em;
      }
    }
  }
}
</style>
<script>
import packageInfo from '@/../package.json'

export default {
  name: 'Footer',
  computed: {
    version () {
      return packageInfo.version
    }
  }
}
</script>
