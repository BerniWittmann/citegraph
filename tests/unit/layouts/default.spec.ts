import { shallowMount } from '@vue/test-utils'
import { i18n } from '../setupPlugins'

import DefaultLayout from '@/layouts/Default.vue'

import Navigation from '@/components/Navigation.vue'
import Footer from '@/components/Footer.vue'

describe('layouts/Default.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(DefaultLayout, {
      i18n,
      slots: {
        default: '<p class="content">Main Content</p>'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('renders the navigation', () => {
    const wrapper = shallowMount(DefaultLayout, {
      i18n
    })
    expect(wrapper.contains(Navigation)).toBeTruthy()
  })
  it('renders the main content', () => {
    const wrapper = shallowMount(DefaultLayout, {
      i18n,
      slots: {
        default: '<p class="content">Main Content</p>'
      }
    })
    const content = wrapper.find('.content')
    expect(content.exists()).toBeTruthy()
    expect(content.text()).toEqual('Main Content')
  })
  it('renders the footer', () => {
    const wrapper = shallowMount(DefaultLayout, {
      i18n
    })
    expect(wrapper.contains(Footer)).toBeTruthy()
  })
})
