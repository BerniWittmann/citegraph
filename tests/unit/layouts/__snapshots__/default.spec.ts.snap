// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts/Default.vue renders 1`] = `
<el-container-stub>
  <el-header-stub height="60px">
    <v-navigation-stub></v-navigation-stub>
  </el-header-stub>
  <el-main-stub>
    <p class="content">Main Content</p>
  </el-main-stub>
  <el-footer-stub height="60px">
    <v-footer-stub></v-footer-stub>
  </el-footer-stub>
</el-container-stub>
`;
